src/scanner.zig

```diff
@@ -253,20 +253,29 @@ pub const Scanner = struct {
     }
 
     fn number(self: *Self) !Token {
-        while (isNumber(self.peek())) {
+        while (isNumber(self.peek()) or self.peek() == '_') {
             _ = self.advance();
         }
 
+        // make sure last digit is not '_'
+        if (self.peek() == '_') {
```

Comment:
peek will look at the next char. If the number ends with underscore you've already consumed it here

Reply:
Thanks, fixed by looking at `self.source[self.current.offset - 1] == '_'` instead.